src/Progressstepper/progressstepper.jsx

```diff
@@ -0,0 +1,228 @@
+      })()
+      :
+      (() => {
+        if (activeStep < currStep) return <StepLight strokeColor={strokeColor} />;
+        if (activeStep > currStep) return <StepDark strokeColor={strokeColor} />;
+        else return <StepActive strokeColor={strokeColor} />;
+      })()
+
+    const getTextColor = maxAvailableStep != undefined ? 
+    () => {
+      if (activeStep === currStep) return 'white';
+      if (maxAvailableStep >= currStep) return 'white';
+      return 'black';
+    }
+    :
+    () => {
+      if (activeStep >= currStep) return 'white';
+      return 'black';
+    }
+
+    return (
+      <Box position="relative" display="inline-flex">
+        {baseIcon}
+        <Typography
+          variant="caption"
+          position="absolute"
+          top="50%"
+          left="50%"
+          sx={{
+            transform: 'translate(-50%, -50%)',
+            color: getTextColor(),
+            fontWeight: activeStep === currStep ? 'bold' : 'normal',
+          }}
+        >
+          {currStep + 1}
+        </Typography>
+      </Box>
+    );
+  };
+
+  return (
+    <Box
+      p="1rem"
+      sx={{
+        backgroundColor: '#F5F5F5',
+        opacity: 0.9,
+        ...boxProps.sx,
+      }}
+      {...boxProps}
+    >
+      <Stepper
+        activeStep={activeStep}
+        alternativeLabel
+        nonLinear
+        connector={<CustomStepConnector />}
+        {...stepperProps}
+      >
+        {steps.map((step, index) => (
+          <Step key={index} completed={index < activeStep} disabled={maxAvailableStep !== undefined ? maxAvailableStep < index : false} {...stepProps}>
+            <StepButton
+              onClick={() => handleStepClick(tabs[index], index)}
+              icon={getStepIcon(index, activeStep)}
+              sx={{
+                '.MuiStepLabel-label': { marginTop: 0 },
+                ...stepButtonProps.sx,
+              }}
+              {...stepButtonProps}
+            >
+              <Typography
+                fontFamily="IBM Plex Sans"
+                color="additional.greydark"
+                sx={{
+                  ...(activeStep === index && {
+                    color: 'main.text',
+                    fontWeight: 700,
+                    textDecoration: 'underline',
+                    textDecorationThickness: '1.5px',
+                    textUnderlinePosition: 'from-font',
+                  }),
+                  ...typographyProps.sx,
+                }}
+                {...typographyProps}
+              >
+                {step}
+              </Typography>
+            </StepButton>
+          </Step>
+        ))}
+      </Stepper>
+    </Box>
+  );
+};
+
+PSAProgressstepper.propTypes = {
+  /**
+   * Array of step labels to display in the stepper.
+   */
+  steps: PropTypes.arrayOf(PropTypes.string).isRequired,
+
+  /**
+   * Array of tabs or additional data that is associated with each step.
+   */
+  tabs: PropTypes.arrayOf(PropTypes.any).isRequired,
```

Comment:
Can you give example about this prop?

Reply:
This prop specifies the tab the app navigates to when the stepButton is clicked.

Reply:
So it's also an array associated with `steps[]` right? I think this might not be necessary to be passed in the component

Reply:
Yup you r right